index.js

```diff
@@ -218,13 +229,13 @@ Pjax.prototype = {
     var state = this.state
 
     if (state.options.history) {
-      if (this.firstrun) {
+      if (!window.history.state) {
         this.lastUid = this.maxUid = newUid()
-        this.firstrun = false
         window.history.replaceState({
             url: window.location.href,
             title: document.title,
-            uid: this.maxUid
+            uid: this.maxUid,
+            scrollPos: 0
```

Comment:
I think you need to reset `scrollPos` to be `[0, 0]` rather than just `0`, so that you're always storing the same type of data against this key.

Reply:
Good point.